mkdocs/docs/bindings/c-api.md

```diff
@@ -1,9 +1,217 @@
 # C API bindings
 
-<!-- TODO(??): overview, advantages/disadvantages to using raw C APIs -->
+IREE provides a low level C API for its runtime[^1], which can be used directly or
+through higher level APIs and language bindings built on top of it.
+
+API header files are organized by runtime component:
+
+| Component header file                                                       | Overview                                                                  |
+|-----------------------------------------------------------------------------|---------------------------------------------------------------------------|
+| [iree/base/api.h](https://github.com/google/iree/blob/main/iree/base/api.h) | Core API, type definitions, ownership policies, utilities                 |
+| [iree/vm/api.h](https://github.com/google/iree/blob/main/iree/vm/api.h)     | VM APIs: loading modules, I/O, calling functions                          |
+| [iree/hal/api.h](https://github.com/google/iree/blob/main/iree/hal/api.h)   | HAL APIs: device management, synchronization, accessing hardware features |
+
+The [iree/samples/](https://github.com/google/iree/tree/main/iree/samples)
+directory demonstrates several ways to use IREE's C API.
 
 ## Prerequisites
 
-<!-- TODO(??): build from source, vcpkg -->
+To use IREE's C API, you will need to build the runtime
+[from source](../building-from-source/getting-started.md). The
+[iree-template-cpp](https://github.com/iml130/iree-template-cpp) community
+project also shows how to integrate IREE into an external project using
+CMake[^2].
+
+## Concepts
```

Comment:
Can we add a diagram that shows the relationship between the components described below?

Reply:
In a follow-up, yeah.